src/internal/_createComposer.js

```diff
@@ -16,7 +17,7 @@ module.exports = function _createComposer(composeFunction) {
                 while (idx--) {
                     fn = composeFunction(arguments[idx], fn);
                 }
-                return arity(length, fn);
+                return curryN(length, arity(length, fn));
```

Comment:
I don't think there's any reason for both `curryN` and `arity`.  You should be able to simply replace `arity` with `curryN` (untested).